### ISSUE TITLE
Set Origin header to the request Origin

### ISSUE THREAD

**Issue by mxmzb** [cyu/rack-cors#109]
I want to allow every Origin to use my API, but it seems I can't do it with the \* sign, cause I need to authorize on it, which ends up telling me the following: `A wildcard '*' cannot be used in the 'Access-Control-Allow-Origin' header when the credentials flag is true.`

So how do I dynamically just send the origin to the request origin?

**Comment by cyu** [comment]
Origin can be a proc, so something like this should work:

```ruby
allow do
  origins ->(origin, env) { origin }
end
```

**Comment by spqpad** [comment]
It doesn't seem to accept the lambda as of version 1.0.0. This is what worked for me:
```
origins do |source, env|
    true
end
```
This proc evaluates the given source, always accepting it, and returning the request origin.

**Comment by cyu** [comment]
@spqpad can you give me an example of what didn't work?  I'll add it to the test.

**Comment by spqpad** [comment]
@cyu Sure. Here's a snippet from the application.rb of the project I was working on:
```ruby
config.middleware.insert_before 0, Rack::Cors do
	allow do
		origins ->(origin, env) { origin }
		resource '*', headers: :any, methods: [:get, :post, :put, :delete], expose: ['Access-Token', 'Client', 'Uid'], credentials: true
	end
end
```

With this config, the server (and workers) fail to start with the error:
`13:27:19 worker.1 | no implicit conversion of Proc into String`

Here's the stacktrace:
```
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:279:in `quote'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:279:in `block in origins'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:273:in `map'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:273:in `origins'
13:27:19 worker.1 | /Users/sab/dev/onemin/config/application.rb:56:in `block (2 levels) in <class:Application>'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:58:in `instance_eval'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:58:in `allow'
13:27:19 worker.1 | /Users/sab/dev/onemin/config/application.rb:51:in `block in <class:Application>'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:43:in `instance_eval'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/rack-cors-1.0.0/lib/rack/cors.rb:43:in `initialize'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.4/lib/action_dispatch/middleware/stack.rb:35:in `new'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.4/lib/action_dispatch/middleware/stack.rb:35:in `build'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.4/lib/action_dispatch/middleware/stack.rb:100:in `block in build'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.4/lib/action_dispatch/middleware/stack.rb:100:in `each'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.4/lib/action_dispatch/middleware/stack.rb:100:in `inject'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/actionpack-5.0.4/lib/action_dispatch/middleware/stack.rb:100:in `build'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/railties-5.0.4/lib/rails/engine.rb:508:in `block in app'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/railties-5.0.4/lib/rails/engine.rb:504:in `synchronize'
13:27:19 worker.1 | /Users/sab/dev/onemin/vendor/bundle/ruby/2.4.0/gems/railties-5.0.4/lib/rails/engine.rb:504:in `app'
```

It seems like it's expecting a string/regex from looking at the source.

**Comment by cyu** [comment]
@spqpad I have a fix (e32b78a).  Will be release in a few.  Thanks!

**Comment by spqpad** [comment]
Thanks a lot!
